src/main/java/com/chessmates/utility/LichessApiImpl.groovy

```diff
@@ -0,0 +1,81 @@
+package com.chessmates.utility
+
+import com.chessmates.model.GameColor
+import com.chessmates.model.Game
+import com.chessmates.model.Player
+import groovy.json.JsonSlurper
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+import java.util.stream.Collectors
+
+/**
+ * Lichess Api.
+ */
+@Component
+class LichessApiImpl implements LichessApi {
+
+    static String LICHESS_API_TEMPLATE = "https://en.lichess.org/api"
+
+    @Autowired
+    HttpUtility utility
+
+    @Override
+    List<Player> getPlayers(String teamId) {
+        // TODO: Handle multiple pages
+        def url = "${LICHESS_API_TEMPLATE}/user?team=${teamId}&nb=50&page=1"
+
+        // TODO: Implement real logging.
+        println "REQUEST: players for team ${teamId}"
+
+        def json = utility.get(url)
+
+        def jsonSlurper = new JsonSlurper()
+        def paginatedResponse = jsonSlurper.parseText(json)
+
+        List<Object> playerObjects = paginatedResponse?.paginator?.currentPageResults
+
+        return playerObjects.stream()
+                .map { playerObject -> parsePlayer(playerObject) }
+                .collect(Collectors.toList())
+    }
+
+    @Override
+    List<Game> getGames(String playerId) {
+        // TODO: Handle multiple pages
+        def url = "${LICHESS_API_TEMPLATE}/user/${playerId}/games?nb=100&page=1"
+
+        println "REQUEST: games for player ${playerId}"
+
+        def json = utility.get(url)
+
+        def jsonSlurper = new JsonSlurper()
+        def paginatedResponse = jsonSlurper.parseText(json)
+
+        // Doesn't have the paginator field in the response for games, unlike players response
+        List<Object> gameObjects = paginatedResponse?.currentPageResults
+
+        return gameObjects.stream()
+                .map { gameObject -> parseGame(gameObject) }
+                .collect(Collectors.toList())
+    }
+
+    private Player parsePlayer(Object playerObject) {
+        new Player(
+                id: playerObject?.id,
+                username: playerObject?.username
+        )
+    }
+
+    private Game parseGame(Object gameObject) {
+        def playerMap = new HashMap<GameColor, String>()
+        playerMap.put(GameColor.WHITE, (String)gameObject?.players?.white?.userId)
```

Comment:
Agreed! I've been quite lax here on error cases. I'm not really concerned about that now though.